util/dvsim/sim_utils.py

```diff
@@ -99,3 +99,76 @@ def vcs_cov_summary_table(buf):
 
     # If we reached here, then we were unable to extract the coverage.
     raise SyntaxError(f"Coverage data not found in {buf.name}!")
+
+
+# Capture and return two runtimes: job_runtime and simulated_time.
+# The run.log file is passed as an input to the function, in addition to the
+# tool used. The tool returns two integers if the runtimes were extracted
+# successfully. The function will raise appropriate exception if the extraction
+# fails.
+def get_runtimes(run_log_txt, tool):
+    with open(run_log_txt, 'r') as f:
+        if tool == 'xcelium':
+            return xcelium_runtimes(f)
+        elif tool == 'vcs':
+            return vcs_runtimes(f)
+        else:
+            # TODO: support using default dvsim run time
+            raise NotImplementedError(f"{tool} is unsupported for run times extraction.")
+
+
+def vcs_runtimes(buf):
+    job_runtime = 0
+    simulated_time = 0
+
+    for line in buf:
+        # Extract job runtime.
+        if "CPU Time" in line:
+            # Find pattern: CPU Time: "XXXX" seconds;.
+            m = re.search(r"(\d+\.\d+)", line)
+            if m:
+                job_runtime = float(m.group(0))
+            else:
+                raise SyntaxError(f"Cannot find job runtime in line: " + line)
+
+        # Extract sim runtime.
+        if "finish at simulation time" in line:
+            # Find pattern: finish at simulation time "XXXX" ps.
+            m = re.search(r"(\d+)", line)
+            if m:
+                simulated_time = int(m.group(0)) / 1_000_000_000.0
+            else:
+                raise SyntaxError(f"Cannot find sim runtime in line: " + line)
+
+        if job_runtime and simulated_time:
+            return job_runtime, simulated_time
+    log.warning("Fail to parse VCS runtimes!")
+
+
+def xcelium_runtimes(buf):
+    job_runtime = 0
+    simulated_time = 0
+
+    for line in buf:
+        # Extract job runtime.
+        if "total:" in line and "TOOL" in line:
+            # Find pattern: TOOL .. (total: "XX":"XX":"XX")
+            m = re.search(r"total:.*(\d+):(\d+):(\d+)", line)
+            if m:
+                # Convert time format: HR:MIN:SEC to seconds.
+                val = re.findall(r"\d+", m.group(0))
+                job_runtime = (int(val[0]) * 60 + int(val[1])) * 60 + int(val[2])
+            else:
+                raise SyntaxError(f"Cannot find job runtime in line: " + line)
```

Comment:
Be sure to handle this exception at the call site too.

Reply:
Also mention it in the doc string.